### PR TITLE
Movie Component

### PR DESCRIPTION
**Pull request by tejas-muthya37**
This PR contains a Movie component that has a functionality of opening a modal to add a Movie to a Playlist and conditional routing based on Auth.

### CODE REVIEW

File: src/Context/products-context.js [hunk 1]
```diff
@@ -0,0 +1,368 @@
...
+    headers: {
+      "Content-Type": "application/json",
+      Accept: "application/json",
+      authorization: encodedToken,
+    },
+    body: JSON.stringify({ video: passedPayload }),
+  })
+    .then((res) => res.json())
+    .then((data) => console.log(data))
+    .catch((err) => console.log(err));
```
Comment (SJTGSHIVAM) [inline src/Context/products-context.js:19]: Please refrain from using console.log instead yo should you some UI related components to show error.

File: src/Context/products-context.js [hunk 2]
```diff
@@ -0,0 +1,368 @@
...
+          if (action.payload.targetElement.checked) {
+            if (!videoFound) {
+              playlist.videos = [...playlist.videos, action.payload.video];
+              addToPlaylistPostRequest(action.payload);
+            }
+          } else {
+            playlist.videos = playlist.videos.filter(
+              (element) => element._id !== action.payload.video._id
+            );
+            removeFromPlaylistDeleteRequest(action.payload);
```
Comment (SJTGSHIVAM) [inline src/Context/products-context.js:291]: almost 29 letters in a name. It's not bad but please remember it should not cross a hard value of 32 at least. please take a note large variable names are good and descriptive but they should also be with a max range of 26-32 for readability you can set that threshold in your mind as per your choice.

File: src/Context/products-context.js [hunk 3]
```diff
@@ -0,0 +1,368 @@
...
+
+  if (dislikedVideosArray === null) dislikedVideosArray = [];
+
+  var watchLaterVideosArray = JSON.parse(
+    localStorage.getItem("WATCH_LATER_ARRAY")
+  );
+
+  if (watchLaterVideosArray === null) watchLaterVideosArray = [];
+
+  var historyVideosArray = JSON.parse(localStorage.getItem("HISTORY_ARRAY"));
```
Comment (SJTGSHIVAM) [inline src/Context/products-context.js:342]: although it's handy to pick up things from local storage but remember local storage is manipulatable and is not reliable as well I remember making a similar mistake. I won't ask for a rewritten login but please validate the details at least before consuming them. Meanwhile it's okay to use anything for a mock application although but be cautious.

File: src/Components/Movie/Movie.js [hunk 4]
```diff
@@ -0,0 +1,165 @@
...
+  const { encodedToken } = useToken();
+
+  useEffect(() => {
+    localStorage.setItem(
+      "PLAYLISTS_ARRAY",
+      JSON.stringify(state.playlistsArray)
+    );
+  }, [state]);
+
+  const style = {
```
Comment (SJTGSHIVAM) [inline src/Components/Movie/Movie.js:29]: Just a quick query are you using something related to styled-components or is it just custom CSS in JS. I know both go hand in hand but was just curious.

**Review by SJTGSHIVAM — APPROVED** [review]
Things look good I have dropped some comments to share your views on them. The code looks clean and readable as well. We are good to go with this one.

File: src/Components/Movie/Movie.js (hunk 4 above)
Comment (tejas-muthya37) [reply src/Components/Movie/Movie.js:29]: This is just the styling for the Modal provided by Material UI

File: src/Components/Movie/Movie.js [hunk 5]
```diff
@@ -0,0 +1,165 @@
+import "./movie.css";
+import { BigPlayButton, Player } from "video-react";
+import WatchLaterIcon from "@mui/icons-material/WatchLater";
+import ThumbUpIcon from "@mui/icons-material/ThumbUp";
+import ThumbDownIcon from "@mui/icons-material/ThumbDown";
+import DeleteIcon from "@mui/icons-material/Delete";
+import PlaylistAddIcon from "@mui/icons-material/PlaylistAdd";
+import CloseIcon from "@mui/icons-material/Close";
+import Modal from "@mui/material/Modal";
+import Box from "@mui/material/Box";
```
Comment (yaswanthmyneni) [inline src/Components/Movie/Movie.js:10]: suggestion: you can create an index file in the assets folder and export all icons from index file and import them here

File: src/Components/Movie/Movie.js [hunk 6]
```diff
@@ -0,0 +1,165 @@
...
+                : props.addToWatchLater();
+            }}
+            fontSize="small"
+          />
+          {props.historyPage && (
+            <DeleteIcon onClick={props.removeFromHistory} fontSize="small" />
+          )}
+          {props.playlistPage && (
+            <DeleteIcon onClick={props.removeFromPlaylist} fontSize="small" />
+          )}
```
Comment (yaswanthmyneni) [inline src/Components/Movie/Movie.js:94]: suggestion: destructure props and use it

File: src/Context/products-context.js [hunk 7]
```diff
@@ -0,0 +1,368 @@
...
+        return {
+          ...state,
+          likedArray: [...state.likedArray, action.payload],
+          dislikedArray: state.dislikedArray.filter(
+            (movie) => movie.id !== action.payload.id
+          ),
+        };
+      }
+    case "Remove from Liked":
+      const dislikedVideoFound = state.dislikedArray.find(
```
Comment (yaswanthmyneni) [inline src/Context/products-context.js:196]: suggestion: write type as ' REMOVE_FROM_LIKED ' instead of "Remove from Liked"

**Review by yaswanthmyneni — APPROVED** [review]
you can merge it

**Pull request merged**